src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -273,102 +224,443 @@ def _is_loop_suitable_for_parallel(self, loop, only_nested_loops=True):
         if only_nested_loops:
             all_loops = loop.walk(Loop)
             if len(all_loops) == 1:
-                self._add_info("Not a nested loop.")
+                self._add_message("Not a nested loop.",
+                                  DACode.INFO_NOT_NESTED_LOOP)
                 return False
 
         if self._loop_types_to_parallelise:
             if loop.loop_type not in self._loop_types_to_parallelise:
-                self._add_info(f"Loop has wrong loop type '{loop.loop_type}'.")
+                self._add_message(f"Loop has wrong loop type '"
+                                  f"{loop.loop_type}'.",
+                                  DACode.INFO_WRONG_LOOP_TYPE)
                 return False
         return True
 
     # -------------------------------------------------------------------------
-    def array_access_parallelisable(self, loop_variable, var_info):
-        '''Tries to determine if the access pattern for a variable
-        given in `var_info` allows parallelisation along the variable
-        `loop_variable`. The following messages might be provided
-        to the user using the message API:
+    @staticmethod
+    def get_flat_indices(component_indices, set_of_loop_vars):
```

Comment:
I don't think the name captures exactly what this method does. Some ideas:

- This is a few levels down the call stack from when we deal with loops. I would call the second parameter just `variables`
- It is a static method that just operates on component_indices. Could this be moved to a method of component_indices?
- It does not completely flatten the structure and the name doesn't convey that it searches only for specific variables. Could this be named something like: `component_indices.get_variables_locations(variables)` or something similar?

Reply:
Done.